### PR TITLE
added class role, and implemented the necessary logic for authentication

### PR DESCRIPTION
**Pull request by RostyslavKuzmych**

### CODE REVIEW

File: src/main/java/application/service/UserServiceImpl.java [hunk 1]
```diff
@@ -7,20 +7,27 @@
...
             throw new RegistrationException("You are already registered");
         }
-        User savedUser = userRepository.save(userMapper.toModel(user));
-        return userMapper.toDto(savedUser);
+        User user = new User();
+        user.setEmail(inputUser.getEmail());
+        user.setPassword(passwordEncoder.encode(inputUser.getPassword()));
+        user.setFirstName(inputUser.getFirstName());
+        user.setLastName(inputUser.getLastName());
+        user.setShippingAddress(inputUser.getShippingAddress());
```
Comment (kshuryhin) [inline src/main/java/application/service/UserServiceImpl.java:30]: Jfyi Lombok offers handy chaining setters as well as builder support

File: src/main/java/application/config/SecurityConfig.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        return new BCryptPasswordEncoder();
+    }
+
+    @Bean
+    public SecurityFilterChain securityFilterChain(HttpSecurity httpSecurity) throws Exception {
+        return httpSecurity
+                .cors(AbstractHttpConfigurer::disable)
+                .csrf(AbstractHttpConfigurer::disable)
+                .authorizeHttpRequests(auth ->
+                        auth.requestMatchers("/auth/**", "/error",
```
Comment (kshuryhin) [inline src/main/java/application/config/SecurityConfig.java:32]: Ok for now, but you can define them in application.properties file and fetch array from there

File: src/main/java/application/repository/UserRepository.java [hunk 3]
```diff
@@ -1,10 +1,13 @@
...
+import java.util.Optional;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.stereotype.Repository;
 
 @Repository
 public interface UserRepository extends JpaRepository<User, Long> {
-    User findByEmail(String email);
+    User getUserByEmail(String email);
+
+    Optional<User> findUserByEmail(String email);
```
Comment (kshuryhin) [inline src/main/java/application/repository/UserRepository.java:12]: Why do we need both of these?

**Review by bhdnchui — APPROVED** [review]
Good job!

**Pull request merged**